### ISSUE TITLE
Crash when parsing a BRIX image

### ISSUE THREAD

**Issue by stevielavern** [LongSoft/Universal-IFR-Extractor#7]
In UEFI.cpp, this line can trigger an out-of-bound vector access when parsing the UEFI form sets:

```cpp
tempFormSet.title = strings[tempFormSet.titleString + stringPackages[tempFormSet.usingStringPackage].structureOffset];
```
This is still applicable to v0.3.3 (503cd54) using the attached file [brix_setup_section.tar.gz](https://github.com/LongSoft/Universal-IFR-Extractor/files/1904073/brix_setup_section.tar.gz). Either the program will crash, or loop forever while filling the disk when writing the output file.

Building with AddressSanitizer highlights the issue; add `set(CMAKE_CXX_FLAGS "-g -fsanitize=address")` in CMakeLists.txt and:

```
$ ./ifrextract brix_setup_section.bin output.txt
Input: brix_setup_section.bin
Output: output.txt
Protocol: UEFI
ASAN:DEADLYSIGNAL
=================================================================
==51086==ERROR: AddressSanitizer: SEGV on unknown address 0x00010df488c0 (pc 0x7fff9dd0f9b0 bp 0x7fff5565ba30 sp 0x7fff5565ba20 T0)
==51086==The signal is caused by a READ memory access.
    #0 0x7fff9dd0f9af in std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::operator=(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) (libc++.1.dylib:x86_64+0x3d9af)
    #1 0x10a64d842 in getUEFIFormSets(std::__1::vector<UEFI_IFR_FORM_SET_PACK, std::__1::allocator<UEFI_IFR_FORM_SET_PACK> >&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::vector<UEFI_IFR_STRING_PACK, std::__1::allocator<UEFI_IFR_STRING_PACK> > const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) UEFI.cpp:204
    #2 0x10a6a3a7b in main main-cli.cpp:160
    #3 0x7fff9f13b254 in start (libdyld.dylib:x86_64+0x5254)

==51086==Register values:
rax = 0x000000000000c208  rbx = 0x00007fff5565bbf8  rcx = 0x000000010df488c0  rdx = 0x000000010a6c67c0
rdi = 0x00007fff5565bbf8  rsi = 0x000000010df488c0  rbp = 0x00007fff5565ba30  rsp = 0x00007fff5565ba20
 r8 = 0x00001000214d8cf8   r9 = 0x0000100000000000  r10 = 0x0000100000000000  r11 = 0x0000100000000000
r12 = 0x00007fff5565ba60  r13 = 0x00007fff5565baa0  r14 = 0x0000100000000000  r15 = 0x000000010a6bc143
AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV (libc++.1.dylib:x86_64+0x3d9af) in std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::operator=(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)
==51086==ABORTING
[1]    51086 abort      ./ifrextract brix_setup_section.bin output.txt
```

Pull request #6 should fix it.

**Comment by NikolajSchlej** [comment]
Fixed in v0.3.4, thanks again.

**Comment by stevielavern** [comment]
Thanks!
